wildmatch_test.go

```diff
@@ -4,6 +4,26 @@ import (
 	"testing"
 )
 
+type PathnameCase struct {
+	Given  string
+	Expect string
+}
+
+func (c *PathnameCase) Assert(t *testing.T) {
+	if got := pathname(c.Given); got != c.Expect {
+		t.Errorf("pathname: expected %s, got %s", c.Expect, got)
+	}
+}
+
+func TestPathname(t *testing.T) {
+	for _, c := range []*PathnameCase{
+		{Given: "*.txt", Expect: "**/*.txt"},
+		{Given: "foo*.txt", Expect: "foo/**/*.txt"},
```

Comment:
MIght also be worth having a case like 
```
{Given: "foo/*/*.txt", Expect: "foo/**/*/**/*.txt"}
```

Reply:
It would also be worth having a case involving a `**` in the path somewhere to make sure that's properly preserved. Also, should a `**/*` pattern expand to `**/**/*`? If so, that might also be worth adding a test case for.